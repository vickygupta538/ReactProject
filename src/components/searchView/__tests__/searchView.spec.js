import React from "react";
import renderer from "react-test-renderer";
import { testStore } from "../../../../test/utils";
import { Provider } from "react-redux";
import SearchView from "../index";

const initialState = {};

describe("SearchView Component Test Cases", () => {
  let store;
  let component;
  beforeEach(() => {
    store = testStore(initialState);
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <SearchView />
      </Provider>
    );
  });

  it("should match snapshot renderer", () => {
    const wrapper = component.toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("simulate change event", () => {
    renderer.act(() => {
      component.root.findByType("input").props.onChange({ target: { value: "test" } });
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
  });
});
