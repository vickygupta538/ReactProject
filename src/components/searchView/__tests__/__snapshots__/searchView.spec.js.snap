// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchView Component Test Cases should match snapshot renderer 1`] = `
<section
  className="searchView"
  data-test="SearchViewComponent"
>
  <div>
    <strong>
      Search:
    </strong>
  </div>
  <input
    className="searchTextBox"
    onChange={[Function]}
    type="input"
  />
</section>
`;
